packages/ember/tests/routing/route_controller_integration_test.js

```diff
@@ -0,0 +1,44 @@
+import {
+  EMBER_NATIVE_DECORATOR_SUPPORT,
+  EMBER_METAL_TRACKED_PROPERTIES,
+} from '@ember/canary-features';
+import { Route } from '@ember/-internals/routing';
+import Controller from '@ember/controller';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
+
+if (EMBER_METAL_TRACKED_PROPERTIES && EMBER_NATIVE_DECORATOR_SUPPORT) {
+  moduleFor(
+    'Route <-> Controller Integration',
+    class extends ApplicationTestCase {
+      async ['@test properties that autotrack the model update when the model changes'](assert) {
+        assert.expect(2);
+
+        this.router.map(function() {
+          this.route('home', { path: '/home/:id' });
+        });
+
+        class HomeRoute extends Route {
+          async model({ id }) {
+            return { value: id, staticValue: 'whatever' };
+          }
+        }
+
+        class HomeController extends Controller {
+          get derivedProperty() {
+            return this.model.value || 'value is unset';
+          }
+        }
+
+        this.add('route:home', HomeRoute);
+        this.add('controller:home', HomeController);
+        this.addTemplate('home', '<h3 class="derivedProperty">{{this.derivedProperty}}</h3>');
+
+        await this.visit('/home/2');
+        assert.equal(this.$('h3').text(), '2', 'the derived property matches the id');
+
+        await this.visit('/home/3');
+        assert.equal(this.$('h3').text(), '3', 'the derived property matches the id');
```

Comment:
do we need jquery selection?

Reply:
probably not, but I saw other tests using it 🤷‍♂

Reply:
should it fail if jquery integration = false?

Reply:
removed -- just using document.querySelector :-\

Reply:
Ya, it’s an internal abstraction that we did during the “drip jQuery requirement” work